### PR TITLE
fix: Checkbox should not show highlight when disabled

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

Kapture 2019-08-08 at 16 46 32

### 💡 Background and solution

Add disabled related style

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix Checkbox highlight by hover when `disabled`.       |
| 🇨🇳 Chinese |    修复 Checkbox 在 `disabled` 时，鼠标悬浮会高亮的问题。       |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
不如把 `visibility: visible;` 的地方加上 `:not(xxxx-disabled)`

**Comment by zombieJ** [comment]
> 不如把 `visibility: visible;` 的地方加上 `:not(xxxx-disabled)`

这么写是为了按状态分，`disabled` 的相关样式都在一起。以后要重构也方便。

**Comment by afc163** [comment]
靠优先级会提高复杂度，也容易互相覆盖，不如精确剪枝。

**Comment by zombieJ** [comment]
从重构看，selector priority 好处比坏处多。遗留代码里的 `not` 很容易变成不敢动的代码。

**Comment by afc163** [comment]
好吧你这么坚持，其实没多大区别。

File: components/checkbox/style/mixin.less [hunk 1]
```diff
@@ -137,6 +137,11 @@
       color: @disabled-color;
       cursor: not-allowed;
     }
+
+    &:hover::after,
+    .@{checkbox-prefix-cls}-wrapper:hover &::after {
+      visibility: hidden;
+    }
```
Comment (afc163) [inline components/checkbox/style/mixin.less]: 另外，这句代码如果不看上下文的样式是看不懂的。

Comment (zombieJ) [reply components/checkbox/style/mixin.less]: 👌我加个备注

**Pull request merged**